pajbot/modules/wide_emote_limit.py

```diff
@@ -0,0 +1,125 @@
+            type="number",
+            required=True,
+            placeholder="",
+            default=15,
+            constraints={"min_value": 1, "max_value": 167},
+        ),
+        ModuleSetting(
+            key="emote_max_width",
+            label="Maximum width of emotes in pixels. Emotes exceeding this width will be counted as wide. For example: Setting this value to 130 means any emote with 131 pixels width or more will be counted as wide",
+            type="number",
+            required=True,
+            placeholder="",
+            default=128,
+            constraints={"min_value": 112, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="bypass_level",
+            label="Level to bypass module",
+            type="number",
+            required=True,
+            placeholder="",
+            default=420,
+            constraints={"min_value": 100, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="moderation_action",
+            label="Moderation action to apply",
+            type="options",
+            required=True,
+            default="Timeout",
+            options=["Delete", "Timeout"],
+        ),
+        ModuleSetting(
+            key="timeout_online",
+            label="Online timeout duration, 0 = off (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="timeout_offline",
+            label="Offline Timeout duration, 0 = off (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="allow_subs_to_bypass",
+            label="Allow subscribers to bypass",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+        ModuleSetting(
+            key="timeout_reason",
+            label="Timeout Reason",
+            type="text",
+            required=False,
+            placeholder="",
+            default="Too many wide emotes",
+            constraints={},
+        ),
+        ModuleSetting(
+            key="disable_warnings",
+            label="Disable warning timeouts",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+    ]
+
+    def on_message(self, source, message, emote_instances, msg_id, **rest):
+        if source.level >= self.settings["bypass_level"] or source.moderator is True:
+            return True
+
+        if self.bot.is_online and self.settings["timeout_online"] == 0:
+            return True
+
+        if not self.bot.is_online and self.settings["timeout_offline"] == 0:
+            return True
```

Comment:
This returns if `timeout_online` and `timeout_offline is set to 0. Which stops the delete message function from working.